classes/slow_course_table.php

```diff
@@ -0,0 +1,162 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_excimer;
+
+/**
+ * Table for displaying grouped profile lists.
+ *
+ * @package   tool_excimer
+ * @author    Jason den Dulk <jasondendulk@catalyst-au.net>
+ * @copyright 2021, Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class slow_course_table extends profile_table {
+
+    /** Columns to be displayed. */
+    const COLUMNS = [
+        'maxduration',
+        'course',
+        'requestcount',
+        'maxcreated',
+        'mincreated',
+        'minduration',
+    ];
+
+    /** @var \moodle_url URL path to use for linking to profile groups. */
+    protected $urlpath;
+
+    /**
+     * Set the URL path to use for linking to profile groups.
+     *
+     * @param \moodle_url $url
+     */
+    public function set_url_path(\moodle_url $url) {
+        $this->urlpath = $url;
+    }
+
+    /**
+     * Sets the SQL for the table.
+     */
+    protected function put_sql(): void {
+        list($filterstring, $filterparams) = $this->get_filter_for_sql();
+
+        $this->set_count_sql(
+            "SELECT count(distinct request)
+            FROM {tool_excimer_profiles}
+            WHERE $filterstring",
+            $filterparams
+        );
+
+        $filterstring .= " AND courseid IS NOT NULL GROUP BY courseid";
+        $this->set_sql(
+            'courseid, count(request) as requestcount, scripttype, max(created) as maxcreated, min(created) as mincreated,
+            max(duration) as maxduration, min(duration) as minduration',
+           '{tool_excimer_profiles}',
+           $filterstring,
+           $filterparams
+        );
+    }
+
+    /**
+     * Get the columns to be displayed.
+     *
+     * @return string[]
+     */
+    protected function get_columns(): array {
+        $columns = self::COLUMNS;
+        if (!$this->is_downloading()) {
+            $columns[] = 'actions';
+        }
+        return $columns;
+    }
+
+    /**
+     * Course column.
+     *
+     * @param \stdClass $record
+     * @return string
+     */
+    public function col_course(\stdClass $record): string {
+        // This should always be set, but still double check.
+        if (empty($record->courseid)) {
+            return '';
+        }
+
+        // Create a drill-down url for this course instead of linking to the course itself.
+        $url = new \moodle_url($this->urlpath, ['courseid' => $record->courseid]);
+        return \html_writer::link($url, helper::course_display_name($record->courseid));
+    }
+
+    /**
+     * Max created column.
+     *
+     * @param \stdClass $record
+     * @return string
+     */
+    public function col_maxcreated(\stdClass $record): string {
```

Comment:
can we inherit a lot of these columns from another table class and some some boilerplate?

Reply:
Ah probably. I did reuse a lot of code from `grouped_profile_table.php` - inheriting it is probably cleaner. Will update...